### DISCUSSION TITLE
Best practice for WZDx -> J2735 message generation

### DISCUSSION THREAD

**Discussion by tedtms** [usdot-jpo-ode/jpo-wzdx#246] [General]
Are there any major pilot examples or discussions happening around how to take an existing WZDx feed (and/or SwzDeviceFeed), extract the relevant location/status/safety information, then generate and broadcast a J2735 TIM or similar connected vehicle message? 

I imagine information like lane closures, worker presence, and work zone start/stop points would be simple enough to convert to J2540 ITIS codes and broadcast via TIM. Are there any other examples? Is there a set of specific feed fields that multiple pilots are extracting, or is everyone doing their own thing?

Thanks!
Ted

**Comment by natedeshmukhtowery** [comment]
Great question. I'd love to hear if anyone is doing or trying this out as well. This brings to mind the open-source Work Zone Data Collection Toolset that FHWA funded recently (https://wzdctool.azurewebsites.net/) but I think that's more around co-generating both a WZDx and Roadside Safety Message feed than in extracting from one to another.
